### PR TITLE
Unblocking `ruby` test from being run.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge
---

###### Summary 

Unblocking `ruby` test from being run. There are still 80+ failing tests, which will be addressed in a future PR (or a set of PRs, if fixing all in one go would make the PR to big).

###### Change Log  

- Added `BuildRequires` on `sudo` and `shado-utils` to allow `ruby` package tests to run.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local package build.

### CODE REVIEW

**Review by rlmenge — APPROVED** [review]

File: SPECS/ruby/ruby.spec [hunk 1]
```diff
@@ -13,6 +13,12 @@ BuildRequires:  openssl-devel
 BuildRequires:  readline
 BuildRequires:  readline-devel
 BuildRequires:  tzdata
+
```
Comment (rlmenge) [inline SPECS/ruby/ruby.spec:16]: Looks like the spec-linter would like the Requires for gmp and openssl above the "if with check". Not sure if this is critical. @thcrain-msft thoughts?

Comment (thcrain-msft) [reply SPECS/ruby/ruby.spec:16]: This doesn't seem to be critical, and Pawel's change is actually more in line with some changes I'm working on for the linter.

Comment (PawelWMS) [reply SPECS/ruby/ruby.spec:16]: @rlmenge: I did these changes on purpose. Like Thomas said, this is closer to what we're going for anyway. The linters is being actively updated to best fit our preferences and needs, so for now I'm not yet in the "blindly trust the linter" mode, but we'd like to get to that point (or close to it) soon.:) For now I'm taking a look at what the linter suggests and pick the things I agree with.

**Review by rlmenge — APPROVED** [review]
:shipit:

**Pull request merged**